app/src/main/java/org/mozilla/focus/utils/SwipeMotionDetector.java

```diff
@@ -0,0 +1,66 @@
+package org.mozilla.focus.utils;
+
+import android.content.Context;
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
+
+public class SwipeMotionDetector implements View.OnTouchListener {
+
+    private GestureDetector gestureDetector;
+    private OnSwipeListener onSwipeListener;
+
+    public SwipeMotionDetector(Context c, OnSwipeListener onSwipeListener) {
+        gestureDetector = new GestureDetector(c, new GestureListener());
+        this.onSwipeListener = onSwipeListener;
+    }
+
+    public boolean onTouch(final View view, final MotionEvent motionEvent) {
+        return gestureDetector.onTouchEvent(motionEvent);
+    }
+
+    private final class GestureListener extends GestureDetector.SimpleOnGestureListener {
+
+        private static final int SWIPE_THRESHOLD = 100;
+        private static final int SWIPE_VELOCITY_THRESHOLD = 100;
+
+        @Override
+        public boolean onDown(MotionEvent e) {
+            return true;
+        }
+
+        @Override
+        public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float velocityY) {
+            boolean result = false;
+            float diffY = e2.getY() - e1.getY();
+            float diffX = e2.getX() - e1.getX();
+            if (Math.abs(diffX) > Math.abs(diffY)) {
```

Comment:
We will honor left/right swipe over up/down swipe if we swipe diagonal, not sure if this is expected and should be documented.